js/utility/bc-google-analytics.js

```diff
@@ -79,13 +79,47 @@ bcpl.utility.googleAnalytics = (() => {
 		addOutboundLinkTracking();
 	};
 
+	const trackEvent = (action, category, label, value) => {
+		const options = {
+			'event_category' : category, 
+			'event_label' : label
+		}
+
+		if (value) {
+			options['value'] = value;
+		}
+
+		gtag('event', action, options);
+	};
+
+	const trackLogin = () => {
+		gtag('event', 'login', { method: 'Google' });
+	};
+
+	const trackSearch = (searchTerm) => {
+		gtag('event', 'search', { search_term: searchTerm });
+	};
+
+	const trackShare = () => {
+		gtag('event', 'share', { method: 'Google' });
```

Comment:
Same as above, the method "Google" might change, we might also want to pass more information like the url, but we'll have to confirm with dan.